ipatests/test_integration/test_installation.py

```diff
@@ -1075,3 +1075,60 @@ def test_replica_install_against_server(self):
                                             self.replicas[0].hostname],
                                            stdin_text=dirman_password)
         assert self.replicas[0].hostname not in cmd.stdout_text
+
+
+class TestIpaJoin(IntegrationTest):
+
+    num_clients = 2
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_dns=False)
+
+    def test_unenroll_option(self):
+        """
+        Test ipa-join Using unenroll option
+        """
+        # install client and kinit admin
+        tasks.install_client(self.master, self.clients[0])
+        tasks.kinit_admin(self.clients[0])
+
+        # checking unenroll option
+        result = self.clients[0].run_command(['ipa-join', '-u',
+                                             self.clients[0].hostname])
+        assert "Unenrollment successful." in result.stderr_text
+
+        # checking hostname option
+        result = self.clients[0].run_command(['ipa-join', '-h',
+                                              self.clients[0].hostname])
+        assert "Keytab successfully retrieved" in result.stderr_text
```

Comment:
The keytab should be verified. Perhaps this would be sufficient:

self.clients[0].run_command(['kinit', '-kt', '/etc/krb5.keytab'])
self.clients[0].run_command(['ipa', 'host-show', self.clients[0].hostname)
# ensure that has_keytab is Yes.